src/ut/connection_pool_test.cpp

```diff
@@ -224,3 +224,37 @@ TEST_F(ConnectionPoolTest, MoveConnectionHandle)
   EXPECT_CALL(conn_pool, destroy_connection(ai_1, 1)).Times(1);
   EXPECT_CALL(conn_pool, destroy_connection(ai_1, 2)).Times(1);
 }
+
+TEST_F(ConnectionPoolTest, FreeOnError)
```

Comment:
As discussed, can you add more comments to this UT - it's quite hard to understand at the moment